anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java

```diff
@@ -261,7 +266,28 @@ private void checkMappedMetricData(MappedMetricData mappedMetricData) {
         if (!updatedDetectors.isEmpty()) {
             log.info("Removed these updated detectors from cache (so they can be reloaded at time of next metric observation): {}", updatedDetectors);
         }
-        synchedTilTime = currentTime;
+        cacheSyncedTillTime = currentTime;
         return updatedDetectors;
     }
+
+    /**
+     * Sync detector last used time to keep a track of detector usage.
+     * On successful sync, update detectorLastUsedSyncedTillTime to currentTime
+     */
+    void detectorLastUsedTimeSync(long currentTime) {
+        long updateDurationInSeconds = (currentTime - detectorsLastUsedSyncedTillTime) / 1000;
+        if (updateDurationInSeconds <= 0 || detectorsLastUsedTimeToBeUpdated.isEmpty()) {
+            return;
+        }
+
+        int queueSize = detectorsLastUsedTimeToBeUpdated.size();
+        //Update last used time only for 1/4th of queue's size at one go for better performance
+        for (int i = 0; i < queueSize / 4; i++) {
+            UUID uuid = detectorsLastUsedTimeToBeUpdated.remove();
+            if (uuid != null) {
+                detectorSource.updatedDetectorLastUsed(uuid);
```

Comment:
but still, we are making a single call per UUID? 
Can we use bulk API call and construct one single request for 1/4 detectors at a time?

Reply:
yeah. I already added a comment to explore on that option. we can do this in second pass.

Reply:
also, update calls are made via scheduler now and not for every metric that comes in.

Reply:
also, update calls are made via scheduler now and not for every metric that comes in.